src/Servers/Kestrel/Transport.Quic/src/Internal/QuicTrace.cs

```diff
@@ -9,21 +9,21 @@ namespace Microsoft.AspNetCore.Server.Kestrel.Transport.Experimental.Quic.Intern
     internal class QuicTrace : IQuicTrace
     {
         private static readonly Action<ILogger, string, Exception?> _acceptedConnection =
-            LoggerMessage.Define<string>(LogLevel.Debug, new EventId(4, nameof(NewConnection)), @"Connection id ""{ConnectionId}"" accepted.");
+            LoggerMessage.Define<string>(LogLevel.Debug, new EventId(1, nameof(NewConnection)), @"Connection id ""{ConnectionId}"" accepted.");
         private static readonly Action<ILogger, string, Exception?> _acceptedStream =
-            LoggerMessage.Define<string>(LogLevel.Debug, new EventId(5, nameof(NewStream)), @"Stream id ""{ConnectionId}"" accepted.");
+            LoggerMessage.Define<string>(LogLevel.Debug, new EventId(2, nameof(NewStream)), @"Stream id ""{ConnectionId}"" accepted.");
```

Comment:
I noticed a QUIC stream overrides `ConnectionId` to be `$"{ConnectionId}:{StreamId}"` and then uses it in places like this as `Stream id ""{ConnectionId}""`.

Couple of issues:
* A little confusing because stream id in QUIC is just a number
* `$"{ConnectionId}:{StreamId}"` allocates a new string per-stream. `ILogger.IsEnabled` would need to be placed around ever place `ConnectionId` is used to avoid the allocation.

Reply:
What would you recommend here? What do we do for HTTP/2 equivalently? Should we just always log both the connection and stream ids separately?

Reply:
Yeah.

Change `QuicStreamContext.ConnectionId` to return its connection's ConnectionId here:

https://github.com/dotnet/aspnetcore/blob/b5ef3f7efe3c3a95fa821f467ee6e112ad370b99/src/Servers/Kestrel/Transport.Quic/src/Internal/QuicStreamContext.cs#L80-L94

Log message before:
`"Stream id ""{ConnectionId}"" accepted."`

Log message after:
`"Connection id ""{ConnectionId}"" stream ID {StreamId} accepted."`

Repeat for the other stream specific log messages.

Reply:
Sounds good.

Reply:
> What do we do for HTTP/2 equivalently?

There's no real equivalent because HTTP/2 streams aren't part of Kestrel's transport abstraction. If QuicStreamContext needs a ConnectionId, which I think it does since it derives from ConnectionContext, shouldn't it have a globally unique ConnectionId? I think we should leave the StreamId as part of QuicStreamContext's ConnectionId, because the "connection" in question isn't the multiplexed connection but the QUIC stream itself.

Reply:
IMO connection is the negotation HTTP/3 connection between the client and server (sending of SETTINGS frames). ConnectionId is an identifer that represents that.

Spicy comment: Is inheriting from `TransportConnection` the abstraction for a `QuicStream`?

Reply:
I was going to suggest a custom version of TransportConnection for the QuicStreamContext as well (and maybe we should still do this), but I realized doesn't solve the issue with the ConnectionId property. The problem is that MultiplexedConnectionContext.AcceptAcync/ConnectAsync return ConnectionContexts to represent the streams, and ConnectionContext.ConnectionId is (in my opinion at least) an identifier that should uniquely identify ConnectionContext.

I agree with @jkotalik that it probably would have been better if ConnectionContext.ConnectionId was renamed to ConnectionContext.Id, but that ship has sailed.

Reply:
> The problem is that MultiplexedConnectionContext.AcceptAcync/ConnectAsync return ConnectionContexts to represent the streams

Why do we have to do that? Could `MultiplexedConnectionContext.AcceptAcync/ConnectAsync` be changed to return a new StreamContext? Or do you think is there just so much overlap between a connection and stream that it makes sense for them to be the same thing?

Reply:
Currently I don't think we have a stream context type. Maybe that's a change we make to Accept/Connect?

Reply:
I think it's worth trying out a StreamContext. It couldn't really derive from anything considering even BaseConnnectionContext has a ConnectionId property, so you lose any polymorphism you might have had. The question becomes is the (Base)ConnectionContext abstraction too leaky to be a worthwhile representation of multiplexed transport streams? It would mean stuff like connection logging middleware, rate limiting middleware, etc... would need to special case ConnectionContext and StreamContext.